src/test/java/org/opensearch/security/dlic/dlsfls/DlsTest.java

```diff
@@ -35,6 +35,13 @@ protected void populateData(Client tc) {
             new IndexRequest("deals").id("1").setRefreshPolicy(RefreshPolicy.IMMEDIATE).source("{\"amount\": 1500}", XContentType.JSON)
         ).actionGet();
 
+        tc.index(
+            new IndexRequest("terms").id("0").setRefreshPolicy(RefreshPolicy.IMMEDIATE).source("{\"foo\": \"bar\"}", XContentType.JSON)
```

Comment:
These new documents have the same id as the previous documents, shouldn't it be different?

Reply:
Unless `terms` index already exists, and has docs with id `0` and `1`, they can have same id as documents in other index, i.e. `deals` index

Reply:
Its a new index so these doc ids are not overlapping with the docs in the `deals` index

Reply:
Good eye @DarshitChanpura